common/src/main/kotlin/com/cosmotech/common/rbac/CsmRbac.kt

```diff
@@ -173,31 +176,54 @@ open class CsmRbac(
     var isAdmin = this.isAdminToken(rbacSecurity)
     if (!isAdmin) {
       val user = getCurrentAccountIdentifier(this.csmPlatformProperties)
-      isAdmin = this.verifyAdminRole(rbacSecurity, user, rolesDefinition)
+      val groups = getCurrentAccountGroups(this.csmPlatformProperties)
+      isAdmin = this.verifyAdminRole(rbacSecurity, user, groups, rolesDefinition)
     }
     return isAdmin
   }
 
   internal fun verifyAdminRole(
       rbacSecurity: RbacSecurity,
-      user: String,
+      user: String?,
+      groups: List<String>?,
       rolesDefinition: RolesDefinition
   ): Boolean {
     logger.debug("RBAC ${rbacSecurity.id} - Verifying if $user has default admin rbac role")
-    val isAdmin = this.getUserRole(rbacSecurity, user) == this.getAdminRole(rolesDefinition)
-    logger.debug("RBAC ${rbacSecurity.id} - $user has default admin rbac role: $isAdmin")
+    var isAdmin = false
+    groups?.forEach {
+      if (this.getUserRole(rbacSecurity, it) == this.getAdminRole(rolesDefinition)) {
+        isAdmin = true
+      }
+    }
+    if (user != null) {
+      if (this.getUserRole(rbacSecurity, user) == this.getAdminRole(rolesDefinition)) {
+        isAdmin = true
+      }
+    }
+    logger.debug("RBAC ${rbacSecurity.id} - $user have default admin rbac role: $isAdmin")
     return isAdmin
   }
 
   internal fun verifyUser(
       rbacSecurity: RbacSecurity,
       permission: String,
       rolesDefinition: RolesDefinition,
-      user: String
+      user: String?,
+      groups: List<String>?
   ): Boolean {
     logger.debug("RBAC ${rbacSecurity.id} - Verifying $user has permission in ACL: $permission")
```

Comment:
`user` might now be null, both start and end logs should be adapted to handle that and/or log group info as well.